File: mac/Sources/DroidMatchCore/RpcWireLimits.swift
/// Numeric limits shared by the framed RPC and transfer data plane.
///
/// These values are part of the documented M1 wire contract. Keep the Android
/// mirror and `docs/protocol*.md` aligned through `tools/check-wire-limits.py`.
public enum RpcWireLimits {
    public static let maximumEnvelopeLengthBytes = 4 * 1024 * 1024
    public static let defaultTransferChunkSizeBytes: UInt32 = 256 * 1024
    public static let maximumTransferChunkSizeBytes = 1024 * 1024
    public static let maximumTransferInFlightChunks = 4
    public static let maximumTransferInFlightBytes = 2 * 1024 * 1024
}

File: android/app/src/main/java/app/droidmatch/m1/RpcWireLimits.java
package app.droidmatch.m1;

/** Numeric limits shared by the framed RPC and transfer data plane. */
final class RpcWireLimits {
    static final int MAX_ENVELOPE_LENGTH_BYTES = 4 * 1024 * 1024;
    static final int DEFAULT_TRANSFER_CHUNK_SIZE_BYTES = 256 * 1024;
    static final int MAX_TRANSFER_CHUNK_SIZE_BYTES = 1024 * 1024;
    static final int MAX_TRANSFER_IN_FLIGHT_CHUNKS = 4;
    static final int MAX_TRANSFER_IN_FLIGHT_BYTES = 2 * 1024 * 1024;

    private RpcWireLimits() {
    }
}
